source/renderer/app/components/staking/delegation-center/DelegationCenterHeader.js

```diff
@@ -1,102 +1,181 @@
+    description: 'Headline for the Delegation center.',
   },
-  descriptionSecondPart: {
-    id: 'staking.delegationCenter.descriptionSecondPart',
-    defaultMessage: '!!!Read',
-    description: 'Delegation description-part2 for the Delegation center.',
+  totalSlots: {
+    id: 'staking.delegationCenter.totalSlots',
+    defaultMessage: '!!!Total slots',
+    description: 'Headline for the Delegation center.',
   },
-  descriptionThirdPart: {
-    id: 'staking.delegationCenter.descriptionThirdPart',
-    defaultMessage: '!!!this article',
-    description: 'Delegation description-part3 for the Delegation center.',
+  headingLeft: {
+    id: 'staking.delegationCenter.headingLeft',
+    defaultMessage: '!!!Cardano epoch {nextEpochStart} starts in',
+    description: 'Headline for the Delegation center.',
   },
-  descriptionFourthPart: {
-    id: 'staking.delegationCenter.descriptionFourthPart',
-    defaultMessage: '!!!to learn how to increase the delegated amount.',
-    description: 'Delegation description-part4 for the Delegation center.',
+  headingRight: {
+    id: 'staking.delegationCenter.headingRight',
+    defaultMessage: '!!!Current Cardano epoch',
+    description: 'Headline for the Delegation center.',
+  },
+  description: {
+    id: 'staking.delegationCenter.description',
+    defaultMessage:
+      'Changes to delegation preferences will take effect from the next epoch.',
+    description: 'Delegation description for the Delegation center.',
   },
 });
 
-type Props = { adaValue: BigNumber, percentage: number };
+type Props = {
+  networkTip: ?TipInfo,
+  nextEpoch: ?NextEpoch,
+};
 
 @observer
 export default class DelegationCenterHeader extends Component<Props> {
   static contextTypes = {
     intl: intlShape.isRequired,
   };
 
-  render() {
+  generateCurrentEpochPanels = (
+    epoch: number,
+    slots: number,
+    totalSlots: number
+  ) => {
     const { intl } = this.context;
-    const { adaValue, percentage } = this.props;
-    const heading = intl.formatMessage(messages.heading);
-    const descriptionSecondPart = intl.formatMessage(
-      messages.descriptionSecondPart
-    );
-    const descriptionThirdPart = intl.formatMessage(
-      messages.descriptionThirdPart
+    const epochLabel = intl.formatMessage(messages.epoch);
+    const currentSlotLabel = intl.formatMessage(messages.currentSlot);
+    const totalSlotsLabel = intl.formatMessage(messages.totalSlots);
+    const labels: Array<string> = [
+      epochLabel,
+      currentSlotLabel,
+      totalSlotsLabel,
+    ];
+
+    const values = [epoch, slots, totalSlots];
+    const keys = ['epoch', 'slots', 'totalSlots'];
+
+    return labels.map<any>((
+      label: string, // eslint-disable-line
+      index: number
+    ) => (
+      <Fragment key={keys[index]}>
+        {this.generateFieldPanel(labels, values, index)}
+      </Fragment>
+    ));
+  };
+
+  generateFieldPanel = (labels: any, values: any, index: number) => {
```

Comment:
@DeeJayElly You can extract this function into outside of component scope (i.e. helper or util, whatever) because this function doesn't use component resource.

Reply:
Fixed